posthog-react-native/src/posthog-rn.ts

```diff
@@ -87,31 +102,32 @@ export class PostHog extends PostHogCore {
       this.flush()
     })
 
-    if (this._persistence === 'file') {
+    if (this._persistence === 'async') {
       if (!storage) {
         console.warn(
           'PostHog was initialised without using PostHog.initAsync - this can lead to race condition issues.'
         )
       }
-
-      this._semiAsyncStorage =
-        storage || new SemiAsyncStorage(options?.customAsyncStorage || buildOptimisiticAsyncStorage())
+    } else {
+      this.setup()
+    }
+    if (this._persistence === 'memory') {
+      this._storage = new PostHogMemoryStorage()
+    } else {
+      this._storage =
+        storage || new PostHogSemiAsyncStorage(options?.customAsyncStorage || buildOptimisiticAsyncStorage())
     }
 
     // Ensure the async storage has been preloaded (this call is cached)
-
-    const setupAsync = async (): Promise<void> => {
-      if (!this._semiAsyncStorage?.isPreloaded) {
-        await this._semiAsyncStorage?.preloadAsync()
-      }
+    const setup = () => {
```

Comment:
moved the await for preLoadAsync to the initAsync, so it also gets called on the constructor for non async storage (memory or customSync).
initAsync should be called for async storage, ie default. But should not, for sync, ie memory, custom storage passed to new PostHog(...., fancyStorage)